### PR TITLE
Fix privacy and imports in krylov solvers

### PR DESCRIPTION
**Pull request by timofeymukha**
This mostly fixed use statements in Krylov solver types, making the use `only`. Also adds `private` at module scope in some places. A little bit of docstrings. No functionality is changed.

### CODE REVIEW

**Review by MartinKarp — APPROVED** [review]
Looks very good. Big improvements over before.

**Review by njansson — COMMENTED** [review]
I would like to try this with recent version of CCE before merging.

File: src/math/ax_helm_fctry.f90 [hunk 1]
```diff
@@ -44,6 +44,9 @@ module ax_helm_fctry
 
 contains
 
+  !> Factory routine for the a Helmholz problem matrix-vector product.
+  !! The selection is based on the compute backend.
+  !! @param Ax The matrix-vector product type to be allocated.
```
Comment (njansson) [inline src/math/ax_helm_fctry.f90:49]:
```suggestion
  !> Factory routine for the a Helmholtz problem matrix-vector product.
  !! The selection is based on the compute backend.
  !! @param Ax The matrix-vector product type to be allocated.
```

File: src/math/bcknd/cpu/ax_helm.f90 [hunk 2]
```diff
@@ -36,13 +36,23 @@ module ax_helm
   implicit none
   private 
 
+  !> CPU matrix-vector product for a Helmholz problem.
```
Comment (njansson) [inline src/math/bcknd/cpu/ax_helm.f90:39]:
```suggestion
  !> CPU matrix-vector product for a Helmholtz problem.
```

File: src/math/bcknd/cpu/ax_helm.f90 [hunk 3]
```diff
@@ -101,6 +111,7 @@ subroutine ax_helm_compute(w, u, coef, msh, Xh)
  
   end subroutine ax_helm_compute
 
+  !> Generic CPU kernel for the Helmholz matrix-vector product.
```
Comment (njansson) [inline src/math/bcknd/cpu/ax_helm.f90:114]:
```suggestion
  !> Generic CPU kernel for the Helmholtz matrix-vector product.
```

File: src/math/ax_helm_fctry.f90 (hunk 1 above)
Comment (njansson) [inline src/math/ax_helm_fctry.f90:49]:
```suggestion
  !> Factory routine for the a Helmholtz problem matrix-vector product.
  !! The selection is based on the compute backend.
  !! @param Ax The matrix-vector product type to be allocated.
```

**Review by njansson — CHANGES_REQUESTED** [review]

**Review by njansson — APPROVED** [review]

**Pull request merged**